Index: alpha.h
===================================================================
RCS file: /afs/cs/project/spin/cvsroot/spin/local/m3cc/gcc/config/alpha/alpha.h,v
retrieving revision 1.1
retrieving revision 1.2
diff -c -r1.1 -r1.2
*** /tmp/T0aaAfva	Wed Nov 13 15:17:58 1996
--- /tmp/T1aaAfva	Wed Nov 13 15:17:58 1996
***************
*** 19,24 ****
--- 19,33 ----
  the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.  */
  
  
+ /* HISTORY
+  * 29-Jul-96  Charles Garrett (garrett) at the University of Washington
+  *	Make profiling calls happen before any stack manipulation and
+  *	 prevent procedures from jumping to the ..ng entry point when
+  *	 profiling. 
+  *
+ 
+ */
+ 
  /* Names to predefine in the preprocessor for this target machine.  */
  
  #define CPP_PREDEFINES "\
***************
*** 514,524 ****
     For the Alpha, `Q' means that this is a memory operand but not a
     reference to an unaligned location.
     `R' is a SYMBOL_REF that has SYMBOL_REF_FLAG set or is the current
!    function.  */
  
  #define EXTRA_CONSTRAINT(OP, C)				\
    ((C) == 'Q' ? GET_CODE (OP) == MEM && GET_CODE (XEXP (OP, 0)) != AND \
!    : (C) == 'R' ? current_file_function_operand (OP, Pmode)	\
     : 0)
  
  /* Given an rtx X being reloaded into a reg required to be
--- 523,535 ----
     For the Alpha, `Q' means that this is a memory operand but not a
     reference to an unaligned location.
     `R' is a SYMBOL_REF that has SYMBOL_REF_FLAG set or is the current
!    function and that profiling is turned off. `R' controls whether or
!    not a call instruction must be followed by a ldgp instruction. */
  
  #define EXTRA_CONSTRAINT(OP, C)				\
    ((C) == 'Q' ? GET_CODE (OP) == MEM && GET_CODE (XEXP (OP, 0)) != AND \
!    : (C) == 'R' ? current_file_function_operand (OP, Pmode) && \
!                   profile_flag == 0	\
     : 0)
  
  /* Given an rtx X being reloaded into a reg required to be
***************
*** 912,917 ****
--- 923,941 ----
     is ever used in the function.  This macro is responsible for
     knowing which registers should not be saved even if used.  */
  
+ 
+ /* SPIN Change */
+ 
+ #define FUNCTION_NAME(FILE)  output_name (FILE)
+ 
+ #define PROFILE_BEFORE_PROLOGUE 
+ 
+ /* DEC C compiler and linker insert profiling instructions before
+    anything else. This will make it easier to patch the return 
+    address of procedures by changing the $ra register before it
+    can be put on the stack. */
+ 
+ 
  #define FUNCTION_PROLOGUE(FILE, SIZE)  output_prolog (FILE, SIZE)
  
  /* Output assembler code to FILE to increment profiler label # LABELNO
***************
*** 923,931 ****
  
  #define FUNCTION_PROFILER(FILE, LABELNO)			\
      do {							\
  	fputs ("\tlda $28,_mcount\n", (FILE));			\
  	fputs ("\tjsr $28,($28),_mcount\n", (FILE));		\
- 	fputs ("\tldgp $29,0($27)\n", (FILE));			\
      } while (0);
  
  
--- 947,955 ----
  
  #define FUNCTION_PROFILER(FILE, LABELNO)			\
      do {							\
+ 	fputs ("\tldgp $29,0($27)\n", (FILE));			\
  	fputs ("\tlda $28,_mcount\n", (FILE));			\
  	fputs ("\tjsr $28,($28),_mcount\n", (FILE));		\
      } while (0);
  
  
